### DISCUSSION TITLE
No "Allow row selecection" on postgres SQL Table

### DISCUSSION THREAD

**Discussion by dado-ca** [Budibase/budibase#14110] [Help] [answered]
Hi,

I have some issues regarding the "Allow row selection" option. Need to export some data to CSV format. I am not using Custom SQL query but rather an SQL Table from PostgreSQL.

So I added an DataProvider, connected the provider to an postgresql table then below the dataprovider I added a table component, and let it point to the above dataprovider. But there is no Allow row selection option

![No allow row selection option](https://github.com/Budibase/budibase/assets/44159556/bfbda536-5b60-4de9-857e-709dcb52b117)

I have couple applications, so here is one of them with the Allow row selection enabled,  I ve followed the same procedure in both cases.

However in the above case I dont see this option. 

I cant figure out what I am doing Wrong.

![Row selection available](https://github.com/Budibase/budibase/assets/44159556/ba6fc2c9-8b32-41d1-83a4-b42f5cd54a5a)

I must mention I am running selfhosted version 2.22.13, the app in which case the allow row selection shows up was developed in one of previous Budibase versrions.

**Comment by ConorWebb96** [comment, marked as answer]
@dado-ca,

You are on a version which likely didn't have this feature included. Can you update to the latest `2.29.12` and see if the checkboxes still don't exist?

**Comment by dado-ca** [comment]
Will try to upgrade. But last time Ineeded to recreate all of the budibase users, for some reasons I wasnt able to export them. 

Thanks for the advice.

**Comment by dado-ca** [comment]
Hi, I upgraded Budibase to the latest version:

![Budibase upgraded 8 7 2024](https://github.com/Budibase/budibase/assets/44159556/155cc736-70e2-4d6b-aa9d-e423033a648a)

After the upgrade, I updated all applications to the new version. The row selection check boxes now appear on all SQL tables by default, even though there is still no "Allow row selection" option in the Table component settings.

However I am now able to add an Button and Export data from the table. Good enough.

Thank you @ConorWebb96 .

![Row selection works](https://github.com/Budibase/budibase/assets/44159556/243b5f1d-c145-4db2-8602-21e5821e03d2)
